### PR TITLE
[#1966] Widget for selecting coordinates

### PR DESCRIPTION
**Pull request by mauriel133**
### Include the HTML in your template:
`{% include "resource-landing-page/coordinates_picker.html" %}`

### Have a container with all the input fields. 

This container should have:
- The class `"hs-coordinates-picker"` if the elements are to be instantiated on page load. Otherwise see below for how to instantiate dynamically.
- The attribute `"data-coordinates-type"` which can be `"point"` or `"rectangle"`
- The input elements contained within (no matter how nested).

The input elements should have:
- The attribute `"data-map-item"` which can be:
-- for a point: `"longitude"` and `"latitude"`
-- for a rectangle: `"northlimit"`,` "eastlimit"`, `"southlimit"` and `"westlimit"`

### Examples
Elements can be instantiated when the page loads. The script will search for the elements with the class "hs-coordinates-picker" and initiate them. Example:
```
<div class="hs-coordinates-picker" data-coordinates-type="point">
   ...
   <input data-map-item="latitude">
   <input data-map-item="longitude">
   ...
</div>
```

The script will run at page load, find the inputs with the specifications above and append the widget to them. It can be used in multiple places in the same page.

Elements can also be instantiated dynamically. Example:
```
<div id="myCoordinates" data-coordinates-type="point">
   ...
   <input data-map-item="latitude">
   <input data-map-item="longitude">
   ...
</div>

<script>
   $(document).ready(function() {
       $("#myCoordinates").coordinatesPicker();
   });
</script>
```

## Screenshots
_The buttons attached to the input elements_
image

_The map modal_
image

### CODE REVIEW

**Comment by pkdash** [comment]
@mauriel133 It seems there is one line of code formatting issue that you want to fix. Is this new functionality (selecting location co-ordinates using the map interface) being used in any resource type yet?

**Comment by mauriel133** [comment]
@pkdash, I'll look into it.

There is a working instance of this in the Resource Specific tab of Timeseries resource.

**Comment by pkdash** [comment]
@mauriel133 Sounds good. If you think it has not broken the timeseries resource. then +1 from me. I will incorporate this functionality into file types once this gets merged to develop. Once you fix the pep8 issue, you can merge this PR.

**Comment by pkdash** [comment]
@mauriel133 See if you can fix the pep8 issue and merge this PR.

**Comment by mauriel133** [comment]
@pkdash, I made changes to the script to include a function to initiate elements dynamically. Please have a look and I will merge if it looks good to you too.

File: theme/static/js/coordinates-picker.js [hunk 1]
```diff
@@ -2,12 +2,88 @@
...
+
+        items.each(function () {
+            var item = $(this);
+
+            // Wrap the element and append a button next to it to trigger the map modal
+            // The button adapts to the size of the input (bootstrap classes: input-sm and input-lg)
+            if (item.hasClass("input-sm")) {
+                item.wrap('<div class="input-group input-group-sm"></div>');
+            }
+            else if (item.hasClass("input-sm")) {
```
Comment (pkdash) [inline theme/static/js/coordinates-picker.js]: The ```else if``` condition is same as the ``if``` condition.

Comment (mauriel133) [inline theme/static/js/coordinates-picker.js]: Oops. Fixed.

**Comment by pkdash** [comment]
@mauriel133 I have left a comment for you to fix.

**Comment by pkdash** [comment]
@mauriel133 Once Jenkins checks are successful you can merge this PR.

**Pull request merged**